src/containers/PartyView.js

```diff
@@ -1,38 +1,55 @@
-import React, { useState, useEffect } from "react";
-import firebase from "firebase";
-import PropTypes from "prop-types";
-import styled from "styled-components";
-import { Fab, Button } from "@material-ui/core";
-import DungeonService from "../services/dungeonService";
+import React, { useState, useEffect } from 'react';
+import firebase from 'firebase';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Fab, Button, TextField } from '@material-ui/core';
+import DungeonService from '../services/dungeonService';
 import {
   SideBar,
   SideBarToggle,
   ContentWithSideBar,
-  RowCenter
-} from "../components/CustomStyled";
-import CharacterSummary from "../components/CharacterSummary";
-import CharacterSheet from "./CharacterSheet";
+  RowCenter,
+} from '../components/CustomStyled';
+import CharacterSummary from '../components/CharacterSummary';
+import CharacterSheet from './CharacterSheet';
 
 function PartyView({ location }) {
   const [sidebar, setSidebar] = useState(false);
   const [idList, setIDList] = useState([]);
   const [characters, setCharacters] = useState([]);
   const [focus, setFocus] = useState(null);
+  const [partyName, setPartyName] = useState('');
 
   function clearInitiative() {
     const { email } = firebase.auth().currentUser;
     characters
-      .forEach(async character => {
-        const { authorized } = await DungeonService.checkUserAuth(character.id, email)
-        if (authorized) DungeonService.saveCharacter({ ...character, initiative: null })
-      })
+      .forEach(async (character) => {
+        const { authorized } = await DungeonService.checkUserAuth(character.id, email);
+        if (authorized) DungeonService.saveCharacter({ ...character, initiative: null });
+      });
+  }
+
+  async function saveParty(event) {
+    event.preventDefault();
+    let parties = localStorage.getItem('parties');
+    parties = parties ? JSON.parse(parties) : {};
+    const characterIds = [];
+    characters.map((character) => {
+      characterIds.push(character.id);
+      // Heroku build fails for some reason without this
+      return null;
+    });
```

Comment:
Build failed because of warning `Line 37:  Expected to return a value in arrow function  array-callback-return`
All array functions need a return value, in this case you are treating `map` as `foreach`.
I would suggest updating lines 36-41 with 

```js
const characterIds = characters.map((character) => character.id);
```

Which should accomplish what you're going for

Reply:
I always forget that arrow functions will return the arrays like that